### DISCUSSION TITLE
Integration of localization platforms

### DISCUSSION THREAD

**Discussion by Bachstelze** [wagtail/wagtail#12501] [Ideas]
Wagtail's internationalization approach is fine for already-finished translations or administrative batches with the Wagtail localize plugin for the admin interface.
It would be neat to integrate other continuous localization platforms like [weblate](https://weblate.org) or [matecat](https://www.matecat.com/) for more features and to suit a translation community.

What is the best approach for such an integration?
Export the PO files from the translation platform and import them into wagtail?

**Comment by NotANormalNerd** [comment]
There is now the https://docs.wagtail.org/en/stable/reference/signals.html#copy-for-translation-done `copy_for_translation` signal available that let's you hook into the "translation" process and export pages. I think you would still have to write your own import and export processes.

**Comment by thibaudcolas** [comment]
Here are similar integrations which I think you’ll be able to learn from, both built on top of wagtail-localize:

- https://github.com/wagtail/wagtail-localize-git
- https://github.com/mozilla/wagtail-localize-smartling

There is also [wagtail-translate](https://github.com/allcaps/wagtail-translate) by @allcaps which has a focus on machine translations.

**Comment by allcaps** [comment]
I'd like to support more translation services in Wagtail Translate. PRs are welcome!

The docs (readme) of Wagtail Translate also shine some light on the way translations evolve in Wagtail: it tries to provide the APIs for third-party packages to add translation apps to Wagtail.

This allows translation solutions to be developed in parallel. Without tying Wagtail to a specific solution. This keeps Wagtail agile, but also allows integrators to choose a solution fitting their project requirements.
